tests/h/search/query_test.py

```diff
@@ -361,19 +362,11 @@ def test_matches_only_annotations_from_specified_group(
 
         result = search.run(webob.multidict.MultiDict({"group": group.pubid}))
 
+        group_service.groupids_readable_by.assert_called_with(
+            pyramid_request.user, filter_=[group.pubid]
+        )
         assert sorted(result.annotation_ids) == sorted(group1_annotations)
```

Comment:
I know this wasn't all added in this PR, but there are some existing oddities in this section of the tests:

Some simplifications:

 * The last two tests, really just seem to be the same test with different emphasis
 * The fact you don't get back some ids in the second one covers the first, so maybe just remove the first?
 * Some calls to `Annotation(groupid="group1").id` which look like they could / should be `Annotation(groupid="group1")`
* I think you can just pass a dict instead of a `webob.multidict.MultiDict` where there are no repeated elements

A bit more of a change:

 * Instead of `group1`, `group2` it would aid readability to give the groups names like "non_matching_group", "user_group", "public_group" etc.
 * `group_service.groupids_readable_by.return_value = [group.pubid]` could be in the fixture
 * This would probably make it possible to have one set of annotations shared between the tests as a fixture

Reply:
> I think you can just pass a dict instead of a webob.multidict.MultiDict where there are no repeated elements

I have a vague recollection that there was a specific reason for using a `MultiDict` here. I think there may have been some API difference vs a `dict`. I've left this as-is for the moment, but I did make several of the other suggested changes:

- I combined the two tests mentioned into one
- I changed `Annotation(...).id` => `Annotation(...)` where the `id` was never used
- I made several group names more meaningful